### PR TITLE
[c++] Final rename `SOMA_READER` -> `SOMA_ARRAY_READER`

### PR DESCRIPTION
**Pull request by johnkerl**
One more following on #1191

### CODE REVIEW

File: libtiledbsoma/include/tiledbsoma/soma_array_reader.h [hunk 1]
```diff
@@ -374,4 +374,4 @@ class SOMAArrayReader {
 
 }  // namespace tiledbsoma
 
-#endif
+#endif  // SOMA_ARRAY_READER
```
Comment (johnkerl) [inline libtiledbsoma/include/tiledbsoma/soma_array_reader.h:377]: idk how people feel about this comment after `#endif` but it's a long-held personal preference of mine ...

https://github.com/johnkerl/scripts/blame/main/fundam/hdrtop

**Review by eddelbuettel — COMMENTED** [review]
:shipit:

**Comment by johnkerl** [comment]
@eddelbuettel :shipit: == 🚢 ?

File: libtiledbsoma/include/tiledbsoma/soma_array_reader.h (hunk 1 above)
Comment (eddelbuettel) [reply libtiledbsoma/include/tiledbsoma/soma_array_reader.h:377]: It's been a long day so I am even more dim-witted than usual.  At the link I see

image

and now I do not know you are talking about.  "Talk to me like I am five."

**Comment by eddelbuettel** [comment]
Yes I learned about :shipit: here years ago.  Apparently it is a thing:

https://www.google.com/search?q=shipit+squirrel+meaning&oq=shipit+squirrel

File: libtiledbsoma/include/tiledbsoma/soma_array_reader.h (hunk 1 above)
Comment (johnkerl) [reply libtiledbsoma/include/tiledbsoma/soma_array_reader.h:377]: @eddelbuettel sorry to be cryptic.

Just an 11-year-old script which shows that every time I create a header file I want to do `:.!hdrtop %` which makes

```
#ifndef FOO_H
#define FOO_H

#endif // FOO_H
```

showing that I've had this stylistic preference for long enough to have automated around it. :)

Comment (eddelbuettel) [reply libtiledbsoma/include/tiledbsoma/soma_array_reader.h:377]: Superflouous at the end of a file like this that has one and only exactly one header guard.

But in principle I like comments at the end of long and nested 'end' or closing brackets.

Lastly for the case of header guards:  the cool kids switched to `#pragma once` to not have this problem

Very lastly too minor an issue to worry.  But if you feel strongly about it and have the urge to touch umpteen files go for it.

**Comment by johnkerl** [comment]
> Yes I learned about :shipit: here years ago. Apparently it is a thing:

Sorry to be cryptic -- I should have said: given that you did :shipit: , can you accept the PR?

**Review by eddelbuettel — APPROVED** [review]

**Comment by johnkerl** [comment]
Thanks @eddelbuettel !!

**Comment by eddelbuettel** [comment]
Yepper!  Sorry about that -- was long day and I meant to 'shipit comment _and_ approve_ and only managed the first part.  Now fixed!

Thanks for cleaning up in any and all dark corners.

**Pull request merged**